cmake/OpenCVUtils.cmake

```diff
@@ -796,11 +796,20 @@ macro(ocv_get_all_libs _modules _extra _3rdparty)
   set(${_3rdparty} "")
   foreach(m ${OPENCV_MODULES_PUBLIC})
     get_target_property(deps ${m} INTERFACE_LINK_LIBRARIES)
+    if(NOT deps)
+      set(deps "")
+    endif()
     list(INSERT ${_modules} 0 ${deps} ${m})
     foreach (dep ${deps} ${OPENCV_LINKER_LIBS})
       if (NOT DEFINED OPENCV_MODULE_${dep}_LOCATION)
         if (TARGET ${dep})
-          list(INSERT ${_3rdparty} 0 ${dep})
+          get_target_property(_isimported ${dep} IMPORTED)
```

Comment:
It is very strange check. There are no correlation between exported targets (you need this) and imported targets. Exported 3rdparty targets are defined via line like this:
```
ocv_install_target(${ZLIB_LIBRARY} EXPORT OpenCVModules ARCHIVE DESTINATION ${OPENCV_3P_LIB_INSTALL_PATH} COMPONENT dev)
```

I'm not sure that we can read contents of `EXPORT OpenCVModules`, but we can update wrapper `ocv_install_target` to build this list.